### PR TITLE
Gas and fees

### PR DESCRIPTION
**Pull request by alcueca**
The gas from each lender is simply calculated by wrapping `flashLoan` in `gasleft()`, and discounting the gas cost of the instructions in the callback.

The fees are extracted from tests and documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,33 +26,36 @@ sequenceDiagram
...
-| [SolidlyWrapper](src/solidly/SolidlyWrapper.sol)              | Aerodrome           | 0x69b6E55f00d908018E2D745c524995bc231D762b | Base                                              |
-| [SolidlyWrapper](src/solidly/SolidlyWrapper.sol)              | Velodrome           | 0xcF13CDdbA3aEf757c52466deC310F221e06238d6 | Optimism                                          |
+## Deployments
+
+| Lender                 | Networks                                          | Address                                    | Gas             | Fees     | Contract                                                      |
+| ---------------------- | ------------------------------------------------- | ------------------------------------------ | --------------- | -------- | ------------------------------------------------------------- |
+| Aave v3                | Arbitrum One, Optimism, Polygon                   | 0x9D4D2C08b29A2Db1c614483cd8971734BFDCC9F2 | 302483          | 0.09%    | [AaveWrapper](src/aave/AaveWrapper.sol)                       |
```
Comment (ultrasecreth) [inline README.md:33]: Aave fees are 0.05%

File: README.md [hunk 2]
```diff
@@ -26,33 +26,36 @@ sequenceDiagram
...
+| Aave v3 (Permissioned) | Ethereum, Gnosis                                  | 0x0c86c636ed5593705b5675d370c831972C787841 | 299756          | 0        | [AaveWrapper](src/aave/AaveWrapper.sol)                       |
+| Spark                  | Ethereum, Gnosis                                  | 0x8cB701df93f2Dae295aE8D7beE5Aa7e4D40CB397 | 302483          | 0.09%    | [AaveWrapper](src/aave/AaveWrapper.sol)                       |
```
Comment (ultrasecreth) [inline README.md:35]: Spark fees are 0%

File: README.md [hunk 3]
```diff
@@ -26,33 +26,36 @@ sequenceDiagram
...
+| Camelot                | Arbitrum One                                      | 0x5E8820B2832aD8451f65Fa2CCe2F3Cef29016D0d | 148686 - 155940 | 0.01%    | [AlgebraWrapper](src/algebra/AlgebraWrapper.sol)              |
+| Algebra + Pendle       | Arbitrum One                                      | 0xC9d66F655b7B35A2B4958bE2FB58E472736Bbc47 | 596718          | 0.01%    | [AlgebraPendleWrapper](src/pendle/AlgebraPendleWrapper.sol)   |
+| Balancer + Pendle      | Arbitrum One                                      | 0xC1Ea6a6df39D991006b39706db7C51f5A1819da7 | 595448          | 0        | [BalancerPendleWrapper](src/pendle/BalancerPendleWrapper.sol) |
+| Aerodrome              | Base                                              | 0x69b6E55f00d908018E2D745c524995bc231D762b | 253829 - 314321 | Variable | [SolidlyWrapper](src/solidly/SolidlyWrapper.sol)              |
+| Velodrome              | Optimism                                          | 0xcF13CDdbA3aEf757c52466deC310F221e06238d6 | 253829 - 314321 | Variable | [SolidlyWrapper](src/solidly/SolidlyWrapper.sol)              |
+
+Contracts are deployed at the same address for the same lender for all its supported networks.
```
Comment (ultrasecreth) [inline README.md:50]: I'd remove this, as it's not true for all the current deployements

File: test/MockBorrower.sol [hunk 4]
```diff
@@ -137,8 +143,11 @@ contract MockBorrower {
         return "";
     }
 
-    function flashBorrow(address asset, uint256 amount) public returns (bytes memory) {
-        return lender.flash(address(loanReceiver), asset, amount, "", this.onFlashLoan);
+    function flashBorrow(address asset, uint256 amount) public returns (bytes memory out) {
+        uint256 startingGas = gasleft();
+        out = lender.flash(address(loanReceiver), asset, amount, "", this.onFlashLoan);
+        uint256 endingGas = gasleft();
```
Comment (ultrasecreth) [inline test/MockBorrower.sol:149]: I think it'd be better to use something like https://github.com/marktoda/forge-gas-snapshot

**Review by ultrasecreth — CHANGES_REQUESTED** [review]

File: test/MockBorrower.sol (hunk 4 above)
Comment (alcueca) [reply test/MockBorrower.sol:149]: I can do that, do you want to block this until I get the time? Happy either way.

**Pull request closed without merging**